src/openrct2/ride/Vehicle.cpp

```diff
@@ -9296,35 +9309,39 @@ int32_t Vehicle::UpdateTrackMotionMiniGolf(int32_t* outStation)
         poweredAcceleration *= powered_acceleration << 1;
         poweredAcceleration = poweredAcceleration / quarterForce;
 
-        if (!(vehicleEntry->flags & VEHICLE_ENTRY_FLAG_WATER_RIDE))
-        {
-            goto loc_6DD054;
-        }
-
-        if (poweredAcceleration < 0)
-        {
-            poweredAcceleration >>= 4;
-        }
-
-        if (vehicleEntry->flags & VEHICLE_ENTRY_FLAG_SPINNING)
+        if (vehicleEntry->flags & VEHICLE_ENTRY_FLAG_WATER_RIDE)
         {
-            spin_speed = std::clamp(spin_speed, VEHICLE_MIN_SPIN_SPEED_WATER_RIDE, VEHICLE_MAX_SPIN_SPEED_WATER_RIDE);
-        }
+            if (poweredAcceleration < 0)
+            {
+                poweredAcceleration >>= 4;
+            }
 
-        if (vehicle_sprite_type != 0)
-        {
-            poweredAcceleration = std::max(0, poweredAcceleration);
             if (vehicleEntry->flags & VEHICLE_ENTRY_FLAG_SPINNING)
             {
-                if (vehicle_sprite_type == 2)
+                spin_speed = std::clamp(spin_speed, VEHICLE_MIN_SPIN_SPEED_WATER_RIDE, VEHICLE_MAX_SPIN_SPEED_WATER_RIDE);
+            }
+
+            if (vehicle_sprite_type != 0)
+            {
+                poweredAcceleration = std::max(0, poweredAcceleration);
+                if (vehicleEntry->flags & VEHICLE_ENTRY_FLAG_SPINNING)
                 {
-                    spin_speed = 0;
+                    if (vehicle_sprite_type == 2)
+                    {
+                        spin_speed = 0;
+                    }
+                }
+            }
+            else
+            {
+                if (abs(velocity) > 0x10000)
```

Comment:
Can be condensed into `else if...`

Reply:
I've modified it again to make it almost identical to the non golf version getting it ready to combine the two.